Runtime/Interaction/Action-based/ActionBasedControllerManager.cs

```diff
@@ -0,0 +1,787 @@
+                }
+            }
+
+            if (teleportXRInteractor == null)
+            {
+                teleportXRInteractor = teleportController.GetComponent<XRRayInteractor>();
+
+                if (teleportXRInteractor == null)
+                {
+                    Debug.LogWarning($"Cannot find {nameof(XRRayInteractor)} component on the Teleport Controller GameObject.", this);
+                }
+            }
+        }
+
+        private void FindUIControllerComponents()
+        {
+            if (uiController == null)
+            {
+                Debug.LogWarning("Missing reference to the UI Controller GameObject.", this);
+                return;
+            }
+
+            if (uiXRController == null)
+            {
+                uiXRController = uiController.GetComponent<XRBaseController>();
+
+                if (uiXRController == null)
+                {
+                    Debug.LogWarning($"Cannot find {nameof(XRBaseController)} component on the UI Controller GameObject.", this);
+                }
+            }
+
+            if (uiLineVisual == null)
+            {
+                uiLineVisual = uiController.GetComponent<XRInteractorLineVisual>();
+
+                if (uiLineVisual == null)
+                {
+                    Debug.LogWarning($"Cannot find {nameof(XRInteractorLineVisual)} component on the UI Controller GameObject.", this);
+                }
+            }
+
+            if (uiXRInteractor == null)
+            {
+                uiXRInteractor = uiController.GetComponent<XRRayInteractor>();
+
+                if (uiXRInteractor == null)
+                {
+                    Debug.LogWarning($"Cannot find {nameof(XRRayInteractor)} component on the UI Controller GameObject.", this);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find and configure the components on the base controller.
+        /// </summary>
+        /// <param name="enable"> Set it true to enable the base controller, false to disable it. </param>
+        private void SetBaseController(bool enable)
+        {
+            FindBaseControllerComponents();
+
+            if (baseXRController != null)
+            {
+                baseXRController.enableInputActions = enable;
+            }
+
+            if (baseXRInteractor != null)
+            {
+                baseXRInteractor.enabled = enable;
+            }
+
+            if (baseXRInteractor is XRRayInteractor && baseXRLineVisual != null)
+            {
+                baseXRLineVisual.enabled = enable;
+            }
+        }
+
+        /// <summary>
+        /// Find and configure the components on the teleport controller.
+        /// </summary>
+        /// <param name="enable"> Set it true to enable the teleport controller, false to disable it. </param>
+        private void SetTeleportController(bool enable)
+        {
+            FindTeleportControllerComponents();
+
+            if (teleportLineVisual != null)
```

Comment:
teleportLineVisual?.enabled = enable

Reply:
`The left-hand side of an assignment must be a variable, property or indexer` 🤷‍♂️
![image](https://user-images.githubusercontent.com/6911992/105698815-b66d7000-5f06-11eb-961b-e0be2e828ae0.png)